sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -0,0 +1,730 @@
+     * indicating the outcome. The work can complete with an error if it cannot initiate the request
+     * to the ProtonJ library or the configured timeout elapses.
+     * <p/>
+     * The work is started once the application is subscribed to the {@link Mono} returned by
+     * {@link DispositionWork#getMono()}; the Mono is terminated upon the work completion.
+     */
+    private static final class DispositionWork extends AtomicBoolean {
+        private final AtomicInteger tryCount = new AtomicInteger(1);
+        private final String deliveryTag;
+        private final DeliveryState desiredState;
+        private final Duration timeout;
+        private Mono<Void> mono;
+        private MonoSink<Void> monoSink;
+        private Instant expirationTime;
+        private Throwable rejectedOutcomeError;
+
+        /**
+         * Create a DispositionWork.
+         *
+         * @param deliveryTag  The delivery tag of the Delivery for which to send the disposition frame requesting
+         *                     delivery settlement on the broker.
+         * @param desiredState The state to include in the disposition frame indicating the desired-outcome
+         *                     the application wish to occur at the broker.
+         * @param timeout      after requesting the ProtonJ library to send the disposition frame, how long to wait for
+         *                     an acknowledgment disposition frame to arrive from the broker.
+         */
+        DispositionWork(String deliveryTag, DeliveryState desiredState, Duration timeout) {
+            this.deliveryTag = deliveryTag;
+            this.desiredState = desiredState;
+            this.timeout = timeout;
+            this.monoSink = null;
+        }
+
+        /**
+         * Gets the delivery tag.
+         *
+         * @return the delivery tag.
+         */
+        String getDeliveryTag() {
+            return deliveryTag;
+        }
+
+        /**
+         * Gets the state indicating the desired-outcome which the application wishes to occur at the broker.
+         * The disposition frame send to the broker includes this desired state.
+         *
+         * @return the desired state.
+         */
+        DeliveryState getDesiredState() {
+            return desiredState;
+        }
+
+        /**
+         * Gets the number of times the work was tried.
+         *
+         * @return the try count.
+         */
+        int getTryCount() {
+            return tryCount.get();
+        }
+
+        /**
+         * Gets the error received from the broker when the outcome of the last disposition attempt
+         * (by sending a disposition frame) happened to be 'Rejected'.
+         *
+         * @return the error in the disposition ack frame from the broker with 'Rejected' outcome,
+         * null if no such disposition ack frame received.
+         */
+        Throwable getRejectedOutcomeError() {
```

Comment:
Please see the [comment](https://github.com/Azure/azure-sdk-for-java/pull/33593#discussion_r1113458234). Also, at the moment, we can scope completion error to the method level instead of the class level, as it represents the terminal state, so we don't need it for anything else other than terminating subscribers.

Reply:
Please see the [comment](https://github.com/Azure/azure-sdk-for-java/pull/33593#discussion_r1113458234) for the thoughts behind naming 'getRejectedOutcomeError'. 

Since completion represents the terminal state, there is no retry but only subscriber termination; so, at the moment, we could scope the completion error to the method level instead of the class level.